libs/configuration/configuration-core/src/main/kotlin/net/corda/libs/configuration/secret/SecretsServiceFactory.kt

```diff
@@ -0,0 +1,15 @@
+package net.corda.libs.configuration.secret
+
+import com.typesafe.config.Config
+
+interface SecretsServiceFactory {
+    /**
+     * Create and instance of [SecretsService] using the provided configuration
+     * or return null if a service of this type cannot be created using the given configuration.
+     *
+     * @param config
+     * @return implementation of [SecretsService] or null if none can be created of this type.
+     */
+    fun create(config: Config): SecretsService?
```

Comment:
```suggestion
    fun create(config: Config): SecretsService
```
and then throw an exception if `SecretsService` cannot be created.

Reply:
If we throw an exception, then we need to catch/use the exception in the logic around finding out which factory is relevant to the configuration.
Isn't that also an anti-pattern?

Reply:
OK, I see your point whereby `SecretServiceFactory` may not recognise the config therefore returning `null` meaning to say "I cannot handle it, please try the next factory avalable."

Reply:
Correct. I considered an alternative interface where you have methods, a `match` that returns a boolean and a `create`, but it would double up the work needing to be done, so wasn't sure if it'd be worth it. But maybe it is if it's a cleaner api?

Reply:
I think it is fine "as is".

Reply:
After discussion, I changed the code so to use an explicit `type` attribute, which feels cleaner.